modules/navier_stokes/src/auxkernels/NSMachAux.C

```diff
@@ -0,0 +1,50 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "NSMachAux.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<NSMachAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+
+  // Mark variables as required
+  params.addRequiredCoupledVar("u", "x-velocity");
+  params.addRequiredCoupledVar("v", "y-velocity");
+  params.addCoupledVar("w", "z-velocity"); // Only required in 3D...
+  params.addRequiredCoupledVar("temperature", "");
+
+  // Parameters with default values
+  params.addRequiredParam<Real>("gamma", "Ratio of specific heats");
+  params.addRequiredParam<Real>("R", "Gas constant.");
+
+  return params;
+}
+
+NSMachAux::NSMachAux(const InputParameters & parameters) :
+    AuxKernel(parameters),
+    _u_vel(coupledValue("u")),
+    _v_vel(coupledValue("v")),
+    _w_vel(_mesh.dimension() == 3 ? coupledValue("w") : _zero),
```

Comment:
@andrsd and I discussed this, and I think in principle these Kernels could be run on lower-dimensional manifolds living in 3D space, possibly with some slight modifications.  Until we make those, however, the `_mesh.dimension()` check is fast and a bit safer than `isCoupled()`.  I do need to add the same thing for the y-component of velocity, however.